### PR TITLE
modify spark-CSV to use NewHadoopRDD

### PR DESCRIPTION
**Pull request by QiangCai**
1 modify spark-CSV  to use CarbonNewHadoopRDD
2 CarbonNewHadoopRDD extend from NewHadoopRDD and override getPartitions method. This method split block to small piece.

### CODE REVIEW

File: hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadNewInputFormat.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.carbondata.hadoop;
+
+import java.io.IOException;
```
Comment (gvramana) [inline hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadNewInputFormat.java]: why two classes for CarbonDataLoadInputFormat, CarbonDataLoadNewInputFormat?

File: hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadInputFormat.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.TextInputFormat;
+
+public class CarbonDataLoadInputFormat extends TextInputFormat {
```
Comment (gvramana) [inline hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadInputFormat.java]: please write the header explaining purpose of this class

File: hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadNewInputFormat.java (hunk 1 above)
Comment (QiangCai) [inline hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadNewInputFormat.java]: CarbonDataLoadInputFormat for hadoop1
CarbonDataLoadNewInputFormat for hadoop2

File: hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadInputFormat.java (hunk 2 above)
Comment (QiangCai) [inline hadoop/src/main/java/org/carbondata/hadoop/CarbonDataLoadInputFormat.java]: fixed

**Comment by ravipesala** [comment]
@QiangCai what is the use of this PR? Why new datasource is created? Can you mention clearly in description.

**Comment by QiangCai** [comment]
@ravipesala  OK

**Comment by QiangCai** [comment]
i will change sparkcsv to use NewHadoopRDD. So i don't need new input format.

**Comment by QiangCai** [comment]
@gvramana @ravipesala please review.

**Comment by gvramana** [comment]
@QiangCai  I think we don't need to create new data source for this. Only adding sc.sparkContext.hadoopConfiguration.set(
 FileInputFormat.SPLIT_MAXSIZE is sufficient before creating RDD. Please confirm.
One more check is required to check inputdatasize/defaultparalism to decide on block size, if  not sufficient blocks as defaultparallelism

**Comment by QiangCai** [comment]
@gvramana  FileInputFormat.SPLIT_MAXSIZE is invalid for spark-csv.  
datasource: spark-csv:
spark-csv
datasource: spark-csv-newapi
spark-csv-newapi

**Comment by QiangCai** [comment]
@gvramana  please review again.

**Comment by gvramana** [comment]
LGTM

**Pull request merged**